plugins/web-terminal/src/components/TerminalComponent/TerminalComponent.tsx

```diff
@@ -128,7 +161,29 @@ export const TerminalComponent = () => {
             type="password"
             variant="outlined"
             inputRef={tokenRef}
+            required
           />
+          <Select
+            name="namespace"
+            inputRef={namespaceRef}
+            defaultValue="openshift-terminal"
+            style={{ display: isHidden ? 'none' : 'block' }}
+          >
+            {namespaces.map(namespace => {
+              return (
+                <MenuItem key={namespace} value={namespace}>
+                  {namespace}
+                </MenuItem>
+              );
+            })}
+          </Select>
+          <p
+            className={classes.centerInfo}
+            style={{ display: isHidden ? 'none' : 'block' }}
+          >
+            Unable to create devworkspace resource in {formData.namespace}{' '}
+            namespace, please choose a different one
+          </p>
```

Comment:
Let's show this only if the user doesn't have access to `openshift-terminal` namespace. The same way as it is in OCP web terminal:

If user has access to `openshift-terminal` let's proceed normally and default to it. If this namespace is not available and creation fails, list additional namespaces and let the user choose.

Reply:
I am not sure if I understand correctly, because it should work like that first time user comes and enters token it tries to use `defaultNamespace` which is either entered through config or by default `openshift-terminal`. If it fails the dialog with dropdown menu is displayed. If the creation succeeds the terminal with environment is displayed